3.2 backports 202502

vadz opened the pull request.

vadz committed:
Avoid using a bashism in script returning CPU count

While ((var++)) works in bash, it results in an error message (and
doesn't increment the variable) in dash, so use a POSIX-compatible
construction which works for both of them.

(cherry picked from commit c29955775a7cbf87280fef60fb106ace2b7d0bb9)

vadz committed:
Deal with the absence of lsb_release in the pre-build script

For now just don't do anything distribution-specific if it's absent, if
we really need it, we can install it later.

(cherry picked from commit b2f186cda1649e4e37efe7e3a6776b402b238faa)

MaartenBent committed:
Use go-httpbin instead of python httpbin for the tests

Python version is too fragile and gets constantly broken by the changes
in Python ecosystem, so switch to the hopefully more stable Go version.

See #23398.

(cherry picked from commit abaeec23bfd97082f4f60f6396ef197d80cf4480)

matyalatte committed:
Fix compilation errors when wxUSE_INTL is off

See #23663.

(cherry picked from commit 77e825f89e0188677a686679d2b1d948c382e381)

vadz committed:
CMake: Set hidden symbol visibility properties

See #24635.

(cherry picked from commit d1506f40c67453ad5ad6799dc6e71a7e71ff56f4)

vadz committed:
Fix logic in wxTextWrapper::Wrap()

The existing code behaved completely wrongly whenever a word longer than
the maximal width was present as it just gave up and didn't wrap
anything at all after it -- when it should have just let this word
overflow but still wrap the rest.

Fix this and add a unit test checking that this works correctly now
(several of the tests would have failed before).

See #23339, #25100.

(cherry picked from commit 597273020037ed853e0864d5bc2c1478a3a51286)

vadz committed:
Fix wxTextWrapper::Wrap() when width is just at the word boundary

Don't move words to the next line if they just fit into the current one.

This requires checking for the space just at the boundary before looking
for the previous or next space and also using the correct predicate with
lower_bound() which uses inexact comparison.

See #25100.

(cherry picked from commit c056a2f4f812f6016e267499e797eb9f2d37d102)

vadz committed:
CMake: Fix using static runtime with msvc and newer CMake

This is a combined cherry-pick of 2c4bd648f2 (Merge branch
'cmake-msvc-static-runtime' of https://github.com/MaartenBent/wxWidgets,
2025-01-21) and cd9acf675d (Another for for using static runtime with
msvc in CMake, 2025-01-21).

See #25098.

taler21 committed:
Correct German translation of "Ctrl" in internat sample

See #25110.

(cherry picked from commit 66c8e32756107eb84988de306406838537d977d7)

sodevel committed:
Add missing wxTextCtrl styles to its XRC handler

Add support for wxTE_CENTER spelling to wxTextCtrl XRC handler, as well
as wxTE_BESTWRAP and wxTE_CAPITALIZE.

See #25118.

(cherry picked from commit bcdf0de4773a024f89221017124c3fc4f343b8fd)

Blake-Madden committed:
Minor wording improvements in wxTextCtrl documentation

See #25119.

(cherry picked from commit fc423ab47fd1d2ec343f8ec16cc8f43777aaba94)

vadz committed:
Fix documentation of wxArrayString ctors taking C arrays

Don't an unnecessary documentation group, this results in these ctors
being documented in a separate documentation section from all the other
ctors.

See #25113.

(cherry picked from commit d1aa0507210d0dfe9b005e552993579690a6f4e2)

swt2c committed:
Add missing documentation for wxBoxSizer::InformFirstDirection()

See #25122.

(cherry picked from commit 428f9b44b056f4a9b3ee3005cb5a55f707acf192)

vadz committed:
Recognize "current" and "pressed" wxBitmapButton buttons too

Recognize not only the old "hover" and "selected" names, but the new and
recommended ones too in wxrc.

See #25114.

(cherry picked from commit 71c7599cd923fc3d4f7d04b6f9e12b87ceaff3af)

taler21 committed:
Improve padding of wxMenu and wxMenuItem in high DPI under MSW

Work around the incorrect behavior of the native control, which does
not scale its padding at high DPI.

This is a backport of f6e31dab42 (Improve padding of wxMenu and
wxMenuItem in high DPI under MSW, 2025-01-24)

See #24821, #25117.

Closes #25137.

vadz committed:
Fix undefining wxDEFINE_UNICHAR_CMP_WITH_INT

It was misspelled, resulting in the original macro staying defined even
when it wasn't supposed to.

Closes #25141.

(cherry picked from commit de1a6be7cb7d0b2da27134b8adfd2c4d68d252bf)

MaartenBent commented:
Can you backport #25098 (and the fix-up commit), or add the backport tag? You asked about it, but it is not yet included.
Looking at my previous PRs, I think #24635 can be backported as well.

vadz commented:
> Can you backport #25098 (and the fix-up commit), or add the backport tag? You asked about it, but it is not yet included. Looking at my previous PRs, I think #24635 can be backported as well.

Both done now, thanks! There was a conflict, but I think I've resolved it correctly ­— please let me know if not.

MaartenBent commented:
Thanks, looks fine. Seems you added `CMP0079` to the policies when resolving the conflict, but that shouldn't hurt anything (it was needed for libcef_dll_wrapper in wx3.3).

vadz committed:
Avoid warning about unused wxFatalExit() in wxMSW DLL build

This function wasn't used when building the library as a DLL, so just
get rid it of entirely and call ExitProcess() directly to avoid warning
C4505 about it being unused and removed from MSVC.

See #25145.

(cherry picked from commit f0937ed038d8e2850872e2686de01a2a47a3fc85)

lanurmi committed:
Add subitem support to wxGenericListCtrl::HitTest()

See #25153.

(cherry picked from commit 580df3d670fa2ee0430588b9a4bc124403b15f86)

Commit:
Fix wrong row parameter in wxEVT_GRID_ROW_MOVE events

Row and column were accidentally exchanged, resulting in row being
always -1 in these events.

wsu-cb committed:
Implement statusbar help text support in wxAuiToolBar

Show the associated "long help" in the associated frame status bar, if
any, just as wxToolBar does.

See #25157.

(cherry picked from commit 0e108e09cdfb6969b0ac73ab43121d571ec1635d)

vadz committed:
Use ubuntu-latest for checks not relying on any specific version

For the checks running simple shell scripts, any Ubuntu version should
work, so switch them to use the latest one to avoid having to update
them every time when an existing version is dropped by GitHub Actions.

(cherry picked from commit 858e2eddc08b45347d449666db6b697c64d26a45)

vadz committed:
Switch a GitHub Actions build from Ubuntu 22.04 to 22.04

Test the latest Ubuntu LTS version too.

(cherry picked from commit 6a914b99a975c2e0e3ccd6864ae581590e132633)

vadz committed:
Run Ubuntu 20.04 CI build using clang in a container

Ubuntu 20.04 will be dropped by GitHub Actions soon, so we can't use it
any more, but switching this build to a newer Ubuntu version doesn't
work because of #25168, so keep running it under 20.04 but use a
container for it.

Fix some check for matrix.container to really test that we're not using
Ubuntu 18.04 container (which is incompatible with the Node.JS version
used by GitHub Actions) instead of testing that we use some container.

(cherry picked from commit 5104110746645669d9b3ffa1e31616fcf5845178)

lanurmi committed:
Include toolkit micro version in wxGetLibraryVersionInfo() too

Instead of outputting just "10.0", output e.g. "10.0.19045".

On Windows, at least, today the micro version has more significance than
it perhaps did 17 years ago when the code was added.

See #25175.

(cherry picked from commit 3fae7f25e4cdfb9a2a94dca99c733656e8000639)

paulcor committed:
Work-around for "drag-end" signal not occurring with Wayland

See #25116

(cherry picked from commit 660849ba608619e31175600aecd6349981b44a9c)

vslavik committed:
wxOSX: Implement clipboard events for single-line wxTextCtrl

Implement wxClipboardEvent sending from wxTextCtrl for NSTextField-based
controls too. Previously they were only sent from NSTextView-based (i.e.
multiline) controls.

See #25165.

(cherry picked from commit aa1f7db472325a831e14ea4a0f3084e4267caed6)

dsa-t committed:
Prevent last pixels cut-off in wxImage::ResampleNearest()

Update the expected test result to match the new (more correct) output.

Closes #25171.

(cherry picked from commit 1415cbf234ed79d5ba0683d3d1cef96851beb6f3)

vadz committed:
Document wxDocManager::GetDocumentManager()

This function exists since always, but somehow was never documented.

Do it now, better late than never.

(cherry picked from commit 8a6efbf5bb91e8fec7f77c4f5096daa663f74520)

vadz committed:
Fix install names of macOS shared libraries

Since the changes of fa2fb36a82 (Fix dylib symlinks on Apple Silicon
Macs, 2024-05-09), see #24524, the install names of the libraries were
incorrect as they were set to the name of the library file instead of
the symlink pointing to it as before. This broke backwards compatibility
as libraries linked with 3.2.5 libraries couldn't find 3.2.6 libraries
which had different names, even though they were still ABI-compatible.

Fix this by simplifying the logic of change-install-names script:
instead of iterating over all libraries, iterate only over the top-level
links, but apply the changes only to the real libraries and not the
symlinks to them.

See #25173, #25182.

vadz committed:
Use FreeBSD 15 in Cirrus CI

FreeBSD 14 is not available any longer, so switch to 15.

This is the equivalent of e439b2347f (Try using FreeBSD 15.0 on Cirrus
CI, 2025-02-11) on master.

vadz committed:
Fix showing labels of newly added wxNotebook pages in wxOSX

The existing code added new tabs before setting their label, which
resulted in them being shown initially truncated.

Fix this by adding new tabs with their correct label (and image), as
this seems to make more sense and also fixes the problem.

Also don't override SetMaximum() in wxCocoaTabView, this doesn't make
sense as wxNotebook doesn't have a "maximum" property unlike wxSpinCtrl
or wxGauge, for which this function (and the associated SetMinimum())
exists, and just do what this function did in SetupTabs() itself, which
is the function explicitly intended to be overridden in wxNotebook.

See #25194, #25196.

(cherry picked from commit 46ff95c84e13db014589982c7399373dd521f6a3)

vadz committed:
Ignore unknown events read from IOCP in wxMSW wxFileSystemWatcher

We can get events with unknown (and, arguably, invalid) values, at least
when using Samba, so don't assert if this happens but just ignore them.

This extends the changes of 83c75a439d (Ignore invalid file notification
events, 2020-10-30) to cover all unknown events and not just those with
"Action" field set to 0.

See #18953.

(cherry picked from commit 380cf411bc7c1e5b3ec82d01656d3ec61e50bf49)
